core/client/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -117,54 +120,54 @@ protected FileInStream(URIStatus status, InStreamOptions options) {
     mFileLength = status.getLength();
     mContext = FileSystemContext.INSTANCE;
     mAlluxioStorageType = options.getAlluxioStorageType();
-    mShouldCacheCurrentBlock = mAlluxioStorageType.isStore();
+    mShouldCache = mAlluxioStorageType.isStore();
     mShouldCachePartiallyReadBlock = options.isCachePartiallyReadBlock();
     mClosed = false;
     mLocationPolicy = options.getLocationPolicy();
-    if (mShouldCacheCurrentBlock) {
+    if (mShouldCache) {
       Preconditions.checkNotNull(options.getLocationPolicy(),
           PreconditionMessage.FILE_WRITE_LOCATION_POLICY_UNSPECIFIED);
     }
-    mSingleByte = new byte[1];
   }
 
   @Override
   public void close() throws IOException {
     if (mClosed) {
       return;
     }
-    if (mShouldCacheCurrentBlock && mShouldCachePartiallyReadBlock) {
-      readCurrentBlockToPos(mFileLength);
+    updateStreams();
+    if (mCurrentCacheStream != null && mShouldCachePartiallyReadBlock) {
+      readCurrentBlockToPos(Long.MAX_VALUE);
```

Comment:
I kinda don't like the way to use `Long.MAX_VALUE` as the parameter to indicate "read the whole block anyway". In this case, can we just make a helper method `readCurrentBlockToEnd()` backed by `readCurrentBlockToPos()`? In that case, our intension is more clear.